lib/tasks/generate_huffman_table.rb

```diff
@@ -119,16 +122,39 @@ class Huffman
 HEADER
         id.times do |i|
           n = id_state[i]
-          f.print "        ["
-          (1 << BITS_AT_ONCE).times do |t|
-            emit = n.transitions[t].emit
-            emit == EOS or emit = emit.dup.force_encoding('binary')
-            f.print %Q/[#{emit == '' ? "nil" : emit.inspect},#{state_id[n.transitions[t].node]}],/
-          end
+          f.print '        ['
+          string = (1 << BITS_AT_ONCE).times.map do |t|
+            transition = n.transitions.fetch(t)
+            emit = transition.emit
+            emit = emit.dup.force_encoding(Encoding::UTF_8) unless emit == EOS
+            emit_literal = case emit
+            when String
+              if emit.empty?
+                'nil'
+              else
+                if emit.valid_encoding?
+                  case emit.bytes
+                  when SINGLE_QUOTE, CARRIAGE_RETURN
+                    emit.inspect
+                  when SLASH
+                    "'\\\\'"
```

Comment:
Why not use `emit.inspect`? `inspect` guarantees it's parsable.
Seems like you are writing code to make the generated code less-readable-but-RoboCop-friendly.